apiserver/params/upgradeseries.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package params
+
+//MachineSeriesUpgradeStatus is the current status a machine series upgrade
+type MachineSeriesUpgradeStatus string
+
+const (
+	MachineSeriesUpgradeNotStarted    MachineSeriesUpgradeStatus = "NotStarted"
+	MachineSeriesUpgradeStarted       MachineSeriesUpgradeStatus = "Started"
+	MachineSeriesUpgradeUnitsRunning  MachineSeriesUpgradeStatus = "UnitsRunning"
+	MachineSeriesUpgradeJujuComplete  MachineSeriesUpgradeStatus = "JujuComplete"
+	MachineSeriesUpgradeAgentsStopped MachineSeriesUpgradeStatus = "AgentsStopped"
+	MachineSeriesUpgradeComplete      MachineSeriesUpgradeStatus = "Complete"
+)
+
+//UnitSeriesUpgradeStatus is the current status of an upgrade
+type UnitSeriesUpgradeStatus string
+
+const (
+	UnitNotStarted UnitSeriesUpgradeStatus = "NotStarted"
+	UnitStarted    UnitSeriesUpgradeStatus = "Started"
+	UnitErrored    UnitSeriesUpgradeStatus = "Errored"
+	UnitCompleted  UnitSeriesUpgradeStatus = "Completed"
+)
```

Comment:
These constants may eventually get replaced with int iota for ordering. This would require a stringer and custom BSON marshaling. Of course, ordering can be done in other ways too.